src/util/amount.rs

```diff
@@ -131,7 +132,7 @@ impl error::Error for ParseAmountError {
 
 
 fn is_too_precise(s: &str, precision: usize) -> bool {
-    s.contains(".") || precision >= s.len() || s.chars().rev().take(precision).any(|d| d != '0')
+    s.contains('.') || precision >= s.len() || s.chars().rev().take(precision).any(|d| d != '0')
```

Comment:
Does this change the `Pattern` from a `&str` to a `char`? Is that more efficient? Neat, though.

Reply:
asm: https://godbolt.org/z/bLK8Hx

The exact code of `Pattern` is complicated by the intuition is as follows:
if char is less than a byte(ie ascii) so you can use `slice::contains` which basically iterates over the slice until it finds it. (`memchr`)
if it's higher then it defaults to the `str` impl which creates a haystack and does a bunch of logic to calculate UTF-8 bytes points